src/ui/utils.js

```diff
@@ -81,8 +81,7 @@ export function getBalloonTablePositionData( editor ) {
  * @returns {module:utils/dom/position~Options}
  */
 export function getBalloonCellPositionData( editor ) {
-	const firstPosition = editor.model.document.selection.getFirstPosition();
-	const modelTableCell = findAncestor( 'tableCell', firstPosition );
+	const modelTableCell = getSelectedTableCell( editor.model.document.selection.getFirstPosition() );
```

Comment:
Why not using the first range and adding there `getSelectedElement()`?

Reply:
If you refer to the `selection.getSelectedElement()` (https://github.com/ckeditor/ckeditor5-engine/blob/bb4def6aac9d221034045f96c1be1f422ff67393/src/model/selection.js#L613-L623) then it is only checking for `nodeAfter` or `nodeBefore` whereas in table UI case we have two cases:

1. A collapsed selection inside `tableCell` (here `findAncestor( position )` works fine
2. A multi-range selection in which ranges are set `'on'` a `tableCell` - here it is the `nodeAfter` first position - but I can lookup API for an existing method on `Range`/`Selection`.

Reply:
If you refer to the `selection.getSelectedElement()` (https://github.com/ckeditor/ckeditor5-engine/blob/bb4def6aac9d221034045f96c1be1f422ff67393/src/model/selection.js#L613-L623) then it is only checking for `nodeAfter` or `nodeBefore` whereas in table UI case we have two cases:

1. A collapsed selection inside `tableCell` (here `findAncestor( position )` works fine
2. A multi-range selection in which ranges are set `'on'` a `tableCell` - here it is the `nodeAfter` first position - but I can lookup API for an existing method on `Range`/`Selection`.